### ISSUE TITLE
DevWorkspace engine MVP

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#19538]
### Is your enhancement related to a problem?

That's a list of subtasks required for the DevWorkspace Engine MVP

### Subtasks

- [x] [Change defaults of chectl/che-operator in case of DevWorkspace](https://github.com/eclipse/che/issues/19218)
- [x] [Projects clone should happen before other components are started](https://github.com/devfile/devworkspace-operator/issues/205)
- [x] Project is not imported when using devfile v2 factory flow #19371
    - [x] che-server API should return the git repo details along with original devfile content https://github.com/eclipse-che/che-server/pull/5
    - [x] dashboard should infer the project details in the original devfile object
- [x] Release devfile spec v2.1 ([milestone](https://github.com/devfile/api/milestone/4)), Che side subtasks:
	- [x] [Commands scoped to a project](https://github.com/devfile/api/issues/165)
	- [x] [Extending DevWorkspace Status fields](https://github.com/devfile/api/issues/370)
	- [x] [Root folders for projects](https://github.com/devfile/api/issues/385)
- [x] Support starting devfile v2 based workspaces on Kubernetes (not OpenShift) **<-- OUT OF SCOPE OF STEP 2**
	- [x] [Support TLS routing on kubernetes](https://github.com/eclipse/che/issues/17853)
	- [x] [Support custom certificates for gateway and workspace endpoints · Issue #19394 · eclipse/che](https://github.com/eclipse/che/issues/19394)
	- [x] Access to Kubernetes API when user is not a Kubernetes user https://github.com/eclipse/che/issues/19813 (we need to discuss if it can be added to the milestone scope)
	- [x] [Support CertManager in che operator for devworkspace controller deployment](https://github.com/eclipse/che/issues/19502)
	- [x] [Support deploying devworkspaces on Kubernetes](https://github.com/eclipse/che/issues/19944)
- [ ] Che-Theia should be able to use a plugin registry that is using an untrusted certificated ==> **to be verify** (it may already work because we are using internal links)
  - [x] Dashboard should provision dashboard and plugin registry URLs into default DWT https://github.com/eclipse/che/issues/19933
- [ ] Che-Theia should be able to download the project zip files even when the devfile registry use an untrusted certificate ==> **OUT OF STEP 2: we don't support airgap**
- [x] WTO and Che with DevWorkspace Engine compatibility (@amisevsk [gdoc proposal](https://docs.google.com/document/d/1-6B1qdizqu-fI8SDwZ0sZslmepN7zQWw7fxrU4Ug35Y/edit#heading=h.778lsysexnzr)) **OpenShift Console will support in 4.8, for CRW we need to discuss the compatibility with all-namespaces**
    - [x] [Prototype DevWorkspace as dependency for WTO and CRW](https://github.com/devfile/devworkspace-operator/issues/407)
- [x] More fine grained startup details on dashboard
	- [x] [Propagate provisioning errors (components/workspaceRoutings)](https://github.com/eclipse/che/issues/17455)
- [x] [Dashboard (or other clients like chectl) should provide metadata about the origin of the devfile with DevWorkspace](https://github.com/eclipse/che/issues/19430)
- [ ] ~~Workspaces authentication~~ (postponed to milestone 3 when we will have #19182)
- [ ] ~~Support for external dependencies repos (e.g. Nexus) that use an untrusted certificate~~
- [ ] ~~Support for proxies that use an untrusted certificate~~
- [ ] ~~Propagation of TLS certificates and other secrets:~~
	- [ ] [How to handle configuration spread across DWO, WTO and DWCO](https://github.com/eclipse/che/issues/19326)
	- [ ] [Add an ability to configure trust stores](https://github.com/eclipse/che/issues/17898)

**Comment by nickboldt** [comment]
Is this issue complete?

**Comment by l0rd** [comment]
I am closing this issue. The out of scope task are tracked by the following issues:

- eclipse/che/issues/20383 related to the recently found downstream bug [https://issues.redhat.com/browse/CRW-2200] 
- eclipse/che/issues/19182
- eclipse/che/issues/17898
- eclipse/che/issues/19326
- [https://github.com/eclipse/che/issues/20386]
- [https://github.com/eclipse/che/issues/20387]
